addons/udes_stock/models/stock_picking.py

```diff
@@ -166,11 +167,15 @@ def get_next_picking_name(self, vals, picking_type=None):
 
         Backorders have a naming pattern of adding `-001` to it,
         so they are more visible to users.
+
+        Do not customise name generation when the picking type differs.
         """
         if vals.get("backorder_id"):
```

Comment:
Question. Is it on the vals picking_type_id of the new picking> If yes maybe smth like this would work i think

```
if vals.get("backorder_id") and vals.get("picking_type_id") == picking_type.id:
    ir_sequence = picking_type.sequence_id
    # Specify sequence as it is picking type specific
    return get_next_name(self, "stock.picking", sequence=ir_sequence)
```

Reply:
we have to check the picking type of the backorder, not the picking type from self in vals. it doesn't fix the issue if we do it this way, unfortunately